pkg/abap/aakaas/package.go

```diff
@@ -0,0 +1,129 @@
+
+// PackageStatus : Status of an ABAP delivery package
+type PackageStatus string
+
+const (
+	// PackageStatusPlanned : Package is Planned
+	PackageStatusPlanned PackageStatus = "P"
+	// PackageStatusLocked : Package is Locked
+	PackageStatusLocked PackageStatus = "L"
+	// PackageStatusReleased : Package is Released
+	PackageStatusReleased PackageStatus = "R"
+	// PackageStatusCreationTriggered : Package was Released but Release procedure is not yet finished
+	PackageStatusCreationTriggered PackageStatus = "C"
+)
+
+type jsonPackageDeterminePackageForScv struct {
+	DeterminePackage struct {
+		Package *Package `json:"DeterminePackageForScv"`
+	} `json:"d"`
+}
+
+type jsonPackage struct {
+	Package *Package `json:"d"`
+}
+
+// Package : ABAP delivery package
+type Package struct {
+	abapbuild.Connector
+	ComponentName       string
+	PackageName         string `json:"Name"`
+	VersionYAML         string
+	Type                string        `json:"Type"`
+	PredecessorCommitID string        `json:"PredecessorCommitId"`
+	Status              PackageStatus `json:"Status"`
+	Namespace           string        `json:"Namespace"`
+}
+
+type packageWithRepository struct {
+	p    Package
+	repo abaputils.Repository
+}
+
+// InitPackage : Initialize Package
+func (p *Package) InitPackage(repo abaputils.Repository, conn abapbuild.Connector) {
+	p.Connector = conn
+	p.ComponentName = repo.Name
+	p.VersionYAML = repo.VersionYAML
+	p.PackageName = repo.PackageName
+	p.Status = PackageStatus(repo.Status)
+}
+
+func (p *Package) copyFieldsToRepo(initialRepo *abaputils.Repository) {
+	initialRepo.PackageName = p.PackageName
+	initialRepo.PackageType = p.Type
+	initialRepo.PredecessorCommitID = p.PredecessorCommitID
+	initialRepo.Status = string(p.Status)
+	initialRepo.Namespace = p.Namespace
+	log.Entry().Infof("Package name %s, type %s, status %s, namespace %s, predecessorCommitID %s", p.PackageName, p.Type, p.Status, p.Namespace, p.PredecessorCommitID)
+}
+
+func (p *Package) reserveNext() error {
+	if p.ComponentName == "" || p.VersionYAML == "" {
+		return errors.New("Parameters missing. Please provide the name and version of the component")
+	}
+	log.Entry().Infof("Reserve package for %s version %s", p.ComponentName, p.VersionYAML)
+	p.Connector.GetToken("/odata/aas_ocs_package")
+	appendum := "/odata/aas_ocs_package/DeterminePackageForScv?Name='" + p.ComponentName + "'&Version='" + p.VersionYAML + "'"
+	body, err := p.Connector.Post(appendum, "")
+	if err != nil {
+		return err
+	}
+	var jPck jsonPackageDeterminePackageForScv
+	json.Unmarshal(body, &jPck)
+	p.PackageName = jPck.DeterminePackage.Package.PackageName
+	p.Type = jPck.DeterminePackage.Package.Type
+	p.PredecessorCommitID = jPck.DeterminePackage.Package.PredecessorCommitID
+	p.Status = jPck.DeterminePackage.Package.Status
+	p.Namespace = jPck.DeterminePackage.Package.Namespace
+	log.Entry().Infof("Reservation of package %s started", p.PackageName)
+	return nil
+}
+
+func (p *Package) get() error {
```

Comment:
The name suggests generic get function, but "only" gets the status and namespace. 
This could be reflected in the function Name -> "getStatusAndNamespace"